### PR TITLE
Added base User functionality

### PR DESCRIPTION
**Pull request by NedelcuRadu**
Login doesn't work for now, need some help.

### CODE REVIEW

**Review by bianca-mihaela-stan — APPROVED** [review]
lgtm

File: src/main/java/ro/unibuc/hello/data/Account.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @NonNull private String lastName;
+    @NonNull private String password;
+    @NonNull private Date dateCreated;
+    private Set<Role> roles = new HashSet<>();
+    public Account(@NotNull String email, @NotNull String firstName, @NotNull String lastName, String password) {
+        this.email = email;
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.dateCreated = new Date();
+        this.password = new BCryptPasswordEncoder().encode(password);
```
Comment (silviabt) [inline src/main/java/ro/unibuc/hello/data/Account.java:29]: 1. this is not good practice to add functionality in constructors
2. and the reason why the /login POST is failing - the Account object is used in the custom implementation of userDetailsService in CustomUserDetailsService.loadUserByUsername(email) => every time an Account gets instantiated the password gets encoded, making the login check to return bad credentials

Comment (silviabt) [reply src/main/java/ro/unibuc/hello/data/Account.java:29]: you could change this `this.password = new BCryptPasswordEncoder().encode(password);` to `this.password = password;` and add the encoding in the part where you add the two users in the db (HelloApplication.runAfterObjectCreated() `Account radu = new Account(".....@gmail.com",
				"Radu", "Nedelcu", passwordEncoder.encode("1234"))`)

Comment (NedelcuRadu) [reply src/main/java/ro/unibuc/hello/data/Account.java:29]: Thanks for the feedback, managed to fix it.

**Pull request merged**